### DISCUSSION TITLE
Initialize persisted directory with the default workspace

### DISCUSSION THREAD

**Discussion by kuldeeparora89** [eclipse-theia/theia-cloud#140] [General]
Hi,

In all the samples I have seen the Theia workspace is not kept under persisted folder. But in real-world scenarios, we would want to preserve the state of the workspace across session restarts. In order to achieve this we tried to copy the workspace from the docker image to the pod "/home/persisted/" using DockerFile. But it is not working,  only the lost+found folder is present under "/home/persisted" & all the workspace content is gone. 

DockerFile snippet -

```
COPY --chown=theia:theia examples/project/ /home/project/persisted/
EXPOSE 3000
ENV SHELL=/bin/bash \
    THEIA_DEFAULT_PLUGINS=local-dir:/home/theia/impaktapps/client_theia_upgrade/browser-app/plugins
ENV USE_LOCAL_GIT true
USER theia

WORKDIR /home/theia/impaktapps/client_theia_upgrade/browser-app
ENTRYPOINT [ "node", "/home/theia/impaktapps/client_theia_upgrade/browser-app/src-gen/backend/main.js" ]
CMD [ "/home/project/persisted", "--hostname=0.0.0.0","--port=3000","--plugins=local-dir:/home/theia/impaktapps/client_theia_upgrade/browser-app/plugins" ]
```

How can we initialize the persisted directory with the default workspace? Do we need to extend the operator to add initContainers or any other preferred approach?

**Comment by jfaltermeier** [comment]
Hi, 

The mount path itself may be influenced via the app definition's `mountPath`, which is set to `/home/project/persisted` in our demos. 
When the pod is launched kubernetes mounts the persisted volume at this path, which should override any data present at this location. So this should be why your Dockerfile approach does not work. 

In production we would usually expect the user to start with an empty workspace initially. 
The Theia application itself may then offer useful ways to initialize the workspace with some sample date (e.g. checking out a git repository or having a Theia Command/Integration on the Getting Started Widget to create some sample or initial data)
So we think this is usually more of an application feature than a Theia Cloud feature. 

We have a ticket about this open (https://github.com/eclipsesource/theia-cloud/issues/69), however this does not have priority at the moment. Such a solution would probably use initContainers however.

**Comment by kuldeeparora89** [comment]
Thanks @jfaltermeier,

We also suspected that it would be due to mount. We will provide a command to user for setting up a project.
